src/apps/competitions/tests/test_featured_competitions.py

```diff
@@ -0,0 +1,59 @@
+from django.test import TestCase
+from django.contrib.auth import get_user_model
+
+from competitions.utils import get_featured_competitions, get_popular_competitions
+from factories import CompetitionFactory, UserFactory, CompetitionParticipantFactory
+
+User = get_user_model()
+
+
+class FeaturedCompetitionsTests(TestCase):
+
+    def setUp(self):
+        """
+        1. Create two different users
+        2. Create few competitions where published is True
+        3. Add participants to competitions
+        3. Return all competitions
+        4. Return competitions with more participants
+        """
+
+        self.user1 = UserFactory()
+        self.user2 = UserFactory()
+        self.user3 = UserFactory()
+        self.user4 = UserFactory()
+        self.user5 = UserFactory()
+        self.user6 = UserFactory()
+        self.user7 = UserFactory()
+
+        self.competition1 = CompetitionFactory(published=True)
+        self.competition2 = CompetitionFactory(published=True)
+        self.competition3 = CompetitionFactory(published=True)
+        self.competition4 = CompetitionFactory(published=True)
+        self.competition5 = CompetitionFactory(published=True)
+        self.competition6 = CompetitionFactory(published=True)
+        self.competition7 = CompetitionFactory(published=True)
+
```

Comment:
Interesting note:
If you don't supply `CompetitionFactory` a `created_by`, it will create a user to satisfy that field, so you have actually created 14 users.
And we aren't actually re-using the users here, so I would refactor this to: 
```python
self.user = UserFactory()  # Create a reusable user
for _ in range(6):  # Create the other 6
    UserFactory()

self.competition1 = CompetitionFactory(create_by=self.user, published=True)
# etc
```
This way we only create the 7 users we need, but we still explicitly know there are 7 users (rather than just implicitly creating them with the CompetitionFactory)

Reply:
Test updated to reflect comment